strawberryfields/ops.py

```diff
@@ -1052,6 +1053,23 @@ def _apply(self, reg, backend, shots=1, **kwargs):
         return backend.measure_fock(reg, shots=shots, select=self.select, **kwargs)
 
 
+class MeasureThreshold(Measurement):
+    """:ref:`photo_detection`: measures a set of modes in the thresholded Fock basis, i.e.
+    zero or nonzero photons.
+
+    After measurement, the modes are reset to the vacuum state.
+    """
+    ns = None
+
+    def __init__(self, select=None):
+        if select is not None and not isinstance(select, Sequence):
+            select = [select]
```

Comment:
How come this extra logic was needed on top of calling the parent `__init__` method?

Reply:
This was copied from `MeasureFock`. This is because the abstract `Measurement` class assumes `select` is always a sequence; at some point, we must have decided that select can be a single number if the measurement applies to a single mode.

Reply:
We should work out how exactly we want the post-selection/measurement API to look, and then refactor this to make it cleaner.

Reply:
@nquesada, even though the Gaussian backend cannot support threshold post-selection, should we add some input validation here to make sure `select` is only 0 or 1?

Reply:
I don't think it's worth it unless we at some decide to write a backend that deals with states that are linear combinations of Gaussian states (this was partially done in https://github.com/xanaduAI/torontonian-sampling).